src/extension_instance.ml

```diff
@@ -167,35 +224,48 @@ end = struct
       LanguageClient.Executable.create ~command ~args ~options ()
   ;;
 
-  let suggest_to_install_ocaml_lsp_server () =
+  let suggest_to_install_ocaml_lsp_server t () =
```

Comment:
I don't think we should modify `suggest_to_install_ocaml_lsp_server` or if we do we should rename it.

It makes sense in an opam or esy sandbox, not when using dune pkg management.

Instead, when the server fails to start in Extension_instance.start_language_server`, we should look at the sandbox: if it is Dune, then propose to run dune lock (if that is the actual correct solution). If it's Opam or another: propose to install ocaml-lsp-server.